### PR TITLE
Fix noreturn discard warning

### LINKED ISSUE #886: Discarded return value warning does not ignore noreturn if defined locally
If I define `noreturn` as `typeof(*null)` in the current file or in an included `object.d` file (using my own custom runtime), then dscanner starts generating warnings that the `noreturn` value is being discarded.

```d
alias noreturn = typeof(*null);

noreturn foo() {
    while (1) {}
}

noreturn bar() {
    foo(); // warning: function value is discarded
}
```

If `noreturn` is not declared locally then it works as expected.

### PR DESCRIPTION
**Pull request by zyedidia**
Fixes #886

### CODE REVIEW

**Review by WebFreak001 — APPROVED** [review]

**Comment by WebFreak001** [comment]
oh, can you add a test?

File: src/dscanner/analysis/unused_result.d [hunk 1]
```diff
@@ -37,12 +37,14 @@ private:
...
 
     const(DSymbol)* void_;
+    const(DSymbol)* noreturn_;
 
     ///
     this(string fileName, const(Scope)* sc, bool skipTests = false)
     {
         super(fileName, sc, skipTests);
         void_ = sc.getSymbolsByName(internString("void"))[0];
+        noreturn_ = sc.getSymbolsByName(internString("noreturn"))[0];
```
Comment (WebFreak001) [inline src/dscanner/analysis/unused_result.d:47]: CI fails here because noreturn might not exist it seems

**Review by WebFreak001 — CHANGES_REQUESTED** [review]

File: src/dscanner/analysis/unused_result.d (hunk 1 above)
Comment (zyedidia) [reply src/dscanner/analysis/unused_result.d]: Thanks, I have added a check to make sure it exists first (and added a unit test).

**Review by WebFreak001 — APPROVED** [review]
while a proper lookup mechanism (e.g. check if it's `typeof(*null)`) would be better, I think this is a reasonable implementation and doing it differently would probably require changes in dsymbol.

**Pull request merged**